javascript/app-model-exposure/src/profile/ExposureAppRoot.tsx

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { RecoilBasedProfileHost } from '@datashaper/app-framework'
+import { memo } from 'react'
+import type { MutableSnapshot, Snapshot } from 'recoil'
+
+import { ModelExposurePage } from '../pages/ModelExposurePage.js'
+import { causalFactorsState } from '../state/causalFactors.js'
+import { causalQuestionState } from '../state/causalQuestion.js'
+import { defaultDatasetResultState } from '../state/defaultDatasetResult.js'
+import { definitionsState } from '../state/definitions.js'
+import { estimatorState } from '../state/estimators.js'
+import { primarySpecificationConfigState } from '../state/primarySpecificationConfig.js'
+import { projectNameState } from '../state/projectName.js'
+import { selectedTableNameState } from '../state/selectedDataPackage.js'
+import type { ExposureResource } from './ExposureResource.js'
+
+export const ExposureAppRoot: React.FC<{
+	resource: ExposureResource
+	href: string
+}> = memo(function ExposureAppRoot({ resource, href }) {
+	return (
+		<RecoilBasedProfileHost
+			resource={resource}
+			saveState={saveState}
+			loadState={loadState}
+		>
+			<ModelExposurePage href={href} />
+		</RecoilBasedProfileHost>
+	)
+})
+
+function loadState(resource: ExposureResource, { set }: MutableSnapshot) {
+	set(projectNameState, resource.projectName)
+	set(causalFactorsState, resource.causalFactors)
```

Comment:
in the current branch I"m working on I added a map to associate an id for the factors, which fixes an error we currently have in the dev site where the table is not being correctly populated. I did the same for the definitions. So my question is, in this version the table pop bug is not happening?

Reply:
I'm not sure TBH, can you demonstrate the issue for me tomorrow?

Reply:
Per our discussion, including a patch in the loadData function